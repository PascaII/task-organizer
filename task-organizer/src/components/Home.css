.wallpaper {
    max-width: 80%;
    max-height: 80%;
    border: 2px solid black;
    border-radius: 7px;
}