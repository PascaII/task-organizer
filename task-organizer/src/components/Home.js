import React from "react";
import "./Home.css";

const Home = () => {
    return (
        <div className="App">
            <div className="container">
                <br />
                <h1>TaskOrganizer</h1>
                <img className="wallpaper" alt="wallpaper" src="home_wallpaper.png" />
                <br />
                <br />
                <div>
                    <a href="/TaskOverview" className="btn btn-primary btn-lg">Go to TaskOverview</a>
                </div>
            </div>
        </div>
    );
};

export default Home;
