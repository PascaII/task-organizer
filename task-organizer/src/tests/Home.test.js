import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from '../components/Home';
import renderer from "react-test-renderer";

test('renders the component without errors', () => {
    render(<Home />);
    expect(screen.getByText('Go to TaskOverview')).toBeInTheDocument();
});

test('renders the image of the component without errors', () => {
    render(<Home />);
    // Use getByRole to query for an image element
    const imageElement = screen.getByRole('img');
    // Expect that the image element is present in the document
    expect(imageElement).toBeInTheDocument();
});

test('snapshot matches implementation', () => {
    const tree = renderer.create(<Home/>).toJSON();
    expect(tree).toMatchSnapshot();
});
