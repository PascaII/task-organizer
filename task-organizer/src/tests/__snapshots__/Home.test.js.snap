// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches implementation 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <br />
    <h1>
      TaskOrganizer
    </h1>
    <img
      alt="wallpaper"
      className="wallpaper"
      src="home_wallpaper.png"
    />
    <br />
    <br />
    <div>
      <a
        className="btn btn-primary btn-lg"
        href="/TaskOverview"
      >
        Go to TaskOverview
      </a>
    </div>
  </div>
</div>
`;
